src/main/java/ca/on/oicr/gsi/dimsum/data/CaseData.java

```diff
@@ -15,12 +16,23 @@ public class CaseData {
   private final List<Case> cases;
   private final Map<String, RunAndLibraries> runsByName;
   private final ZonedDateTime timestamp;
+  private final Set<String> AssayNames;
+  private final Set<String> RequisitionNames;
+  private final Set<String> ProjectsNames;
+  private final Set<String> DonorNames;
+
+
 
   public CaseData(List<Case> cases, Map<String, RunAndLibraries> runsByName,
-      ZonedDateTime timestamp) {
+      ZonedDateTime timestamp, Set<String> Assays, Set<String> Requisitions,
+      Set<String> Projects, Set<String> Donors) {
     this.cases = unmodifiableList(cases);
     this.runsByName = Collections.unmodifiableMap(runsByName);
     this.timestamp = requireNonNull(timestamp);
+    this.AssayNames = Assays;
+    this.RequisitionNames = Requisitions;
+    this.ProjectsNames = Projects;
+    this.DonorNames = Donors;
```

Comment:
Use `Collections.unmodifiableSet` to make these immutable (similar to `unmodifiableMap` used above)